lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,45 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
+    if key_is_duplicated?(method_name)
+      raise "Keys #{method_name} as string and as symbol are present"
+    elsif self.keys.include? method_name
+      self[method_name] = self[method_name].call
+    else
+      raise "Method #{method_name} is missing"
+    end
+  end
+
+  def lazy?(key)
+    key = get_correct_key(key)
+    self[key].respond_to? :call
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    method_name = get_correct_key(method_name)
+    self.keys.include? method_name || super
+  end
+
+  private
+  def get_correct_key(hash_key)
+    [hash_key.to_s, hash_key.to_sym].select { |key| !self[key].nil? }[0]
+  end
+
+  def key_is_duplicated?(hash_key)
+    self.keys.include?(hash_key.to_s && hash_key.to_sym)
```

Comment:
Are you sure that it works correctly?
`[:a, 'b'].include?('c' && :a) # => true`
What do you try to do by `&&`?